packages/ember-model/lib/model.js

```diff
@@ -88,7 +88,7 @@ Ember.Model = Ember.Object.extend(Ember.Evented, {
 
     if (!reference) {
       reference = this.constructor._getOrCreateReferenceForId(id);
-      reference.record = this;
+      set(reference, 'record', this);
       this._reference = reference;
```

Comment:
There is a local variable defined in the ember-model source and thats seems to be the preferred choice in that library.

```
var set = Ember.set();
```